### PR TITLE
Include keywords with only digits.

### PR DESCRIPTION
**Pull request by terw-dan**
Fixes #https://github.com/Yoast/wordpress-seo/issues/3287

Keywords/phrases consisting of digits, or combined with only-digits words where omitted from the keyword matching on the bodytext. 
Removed the replacement  'remove "words" comprised only of numbers' from the cleantext function and moved this to a function in the stringhelper (where it is used for the wordcounts and sentencecounts).  Now the keywords in the text match and the keyworddensity is correct.

### CODE REVIEW

File: js/preprocessor.js [hunk 1]
```diff
@@ -34,29 +34,34 @@ YoastSEO.PreProcessor.prototype.textFormat = function() {
 	this.__store.cleanText = this.cleanText( this.__store.originalText );
 	this.__store.cleanTextSomeTags = this.stringHelper.stripSomeTags( this.__store.cleanText );
 	this.__store.cleanTextNoTags = this.stringHelper.stripAllTags( this.__store.cleanTextSomeTags );
+	this.__store.cleanTextNoDigits = this.stringHelper.stripNumbers( this.__store.cleanTextNoTags );
```
Comment (CarolineGeven) [inline js/preprocessor.js]: We should not strip digits from the text count

File: js/preprocessor.js [hunk 2]
```diff
@@ -34,29 +34,34 @@ YoastSEO.PreProcessor.prototype.textFormat = function() {
...
 /**
  * saves wordcount (all words) and wordcountNoTags (all words except those in tags) in the __store
+ * saves sentencecount and syllable count in __store
  * object
  */
 YoastSEO.PreProcessor.prototype.countStore = function() {
 
 	/*wordcounters*/
-	this.__store.wordcount = this.__store.cleanText === "" ?
+	var wordcountString = this.stringHelper.stripNumbers( this.__store.cleanText );
```
Comment (CarolineGeven) [inline js/preprocessor.js]: We should not strip digits from the text count

File: js/preprocessor.js [hunk 3]
```diff
@@ -34,29 +34,34 @@ YoastSEO.PreProcessor.prototype.textFormat = function() {
...
+
+	this.__store.wordcount = wordcountString === "" ?
 		0 :
-		this.__store.cleanText.split( /\s/g ).length;
+		wordcountString.split( /\s/g ).length;
+
+	var wordcountStringNoTags = this.stringHelper.stripNumbers( this.__store.cleanTextNoTags );
```
Comment (CarolineGeven) [inline js/preprocessor.js]: We should not strip digits from the text count

File: js/preprocessor.js [hunk 4]
```diff
@@ -34,29 +34,34 @@ YoastSEO.PreProcessor.prototype.textFormat = function() {
...
 
-	this.__store.wordcountNoTags = this.__store.cleanTextNoTags === "" ?
+	this.__store.wordcountNoTags = wordcountStringNoTags === "" ?
```
Comment (CarolineGeven) [inline js/preprocessor.js]: For readability please do not use shorthand ifs

File: js/stringhelper.js [hunk 5]
```diff
@@ -124,6 +124,24 @@ YoastSEO.StringHelper.prototype.stripAllTags = function( textString ) {
 };
 
 /**
+ * removes all words comprised only of numbers.
+ * @param textString {String}
+ * @returns {string}
+ */
+YoastSEO.StringHelper.prototype.stripNumbers = function( textString ) {
```
Comment (CarolineGeven) [inline js/stringhelper.js]: This function is not necessary anymore

File: spec/keywordCountSpec.js [hunk 6]
```diff
@@ -43,4 +43,43 @@ describe("a keyword with diacritics", function(){
 		var result = textAnalyzer.keywordCount();
 		expect(result).toBe(1);
 	});
-});
\ No newline at end of file
+});
```
Comment (CarolineGeven) [inline spec/keywordCountSpec.js]: Tests should be fixed to also match on digits

**Comment by atimmer** [comment]
CR Done 👍🏻

**Comment by CarolineGeven** [comment]
## Acceptance
When having digits separated with for example a `.` the focus keyword cannot be found anymore. See print screen.

schermafbeelding_2015-11-19_om_13_42_30

**Comment by CarolineGeven** [comment]
## Acceptance
Digits are not recognized in the URL 
schermafbeelding_2015-11-19_om_14_26_23

**Comment by CarolineGeven** [comment]
Acceptance :+1:

**Pull request merged**